File: examples/java/sdk-reference/observability/step-context-logger.java
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;
import software.amazon.lambda.durable.StepContext;

public class StepContextLogger extends DurableHandler<Object, String> {
    @Override
    public String handleRequest(Object event, DurableContext context) {
        return context.step("process", String.class, (StepContext stepCtx) -> {
            // stepCtx.getLogger() includes operationId, operationName, and attempt in MDC.
            stepCtx.getLogger().info("Running step");
            return "done";
        });
    }
}

File: examples/csharp/sdk-reference/observability/step-context-logger.cs
using Amazon.Lambda.Core;
using Amazon.Lambda.DurableExecution;
using Microsoft.Extensions.Logging;

public class StepContextLoggerExample
{
    public Task<DurableExecutionInvocationOutput> Handler(
        DurableExecutionInvocationInput input, ILambdaContext context)
        => DurableFunction.WrapAsync<object, string>(Workflow, input, context);

    private async Task<string> Workflow(object input, IDurableContext ctx)
    {
        return await ctx.StepAsync(
            async (stepCtx, _) =>
            {
                // stepCtx.Logger includes operationId, operationName, and attempt
                // in every log entry from this step.
                stepCtx.Logger.LogInformation("Running step");
                return "done";
            },
            name: "process");
    }
}
